src/main/scala/piconot/external/parser/Parser.scala

```diff
@@ -0,0 +1,26 @@
+package piconot.external.parser
+import scala.util.parsing.combinator._
+import piconot.external.ir._
+
+object PiconotParser extends JavaTokenParsers with RegexParsers{
+    def apply(s: String): ParseResult[List[OurRule]] = {
+      parseAll(ourrules, s)
+    }
+    
+    def ourrules: Parser[List[OurRule]] = {
+      (   ourstates~"->"~ouractions~";"~ourrules ^^ {case s~"->"~a~";"~more => (List(OurRule(s,a)) ++ more) }
```

Comment:
You might be able to say this a tiny bit more idiomatically by using cons:
`List(OurRule(s, a)) ++ more == OurRule(s, a) :: more`